lib/shopify_api/session.rb

```diff
@@ -143,6 +144,16 @@ def api_version=(version)
       end
     end
 
+    def access_scopes=(access_scopes)
+      return unless access_scopes
+
+      @access_scopes = begin
+        ShopifyAPI::ApiAccess.new(access_scopes)
+      rescue
```

Comment:
I wouldn't do this. Some other (better) options imo, not mutually exclusive:
1. Raise the error when the caller tries to access `access_scopes`, not when we set it. No use raising if it's never even accessed.
2. Raise the error when we set for the caller to handle.
3. Make `valid?` return false when the scopes are bad.

Reply:
1. This is so that we intentionally fail if anyone tries to get `access_scopes`?
2. If `access_scopes` are not passed in, then this should be nil. 
3. I don't think it should validate on `scopes` for this version because this is going to break existing apps.

Reply:
Sorry I think I'm misunderstanding something. The tests imply that this will fail if the scopes are "bad": https://github.com/Shopify/shopify_api/pull/843/files#diff-5e62e8ce4dc1b35b81a888833c02f9a424e3b8eb753fb2301bd8dc736a2eedfcR106

Wouldn't the caller want to know the difference between "scopes are bad" and "no scopes"?

Reply:
ohh yes you're right!

Reply:
I addressed the bad input cases and making sure that `valid?` handles it for access scopes as well.